### PR TITLE
fix(files): remove focus outline on file list item click and make the link focusable

### LINKED ISSUE #41490: [Bug]: Clicking folder names for navigation shows black border for a split second
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

When I click on a folder with my mouse

https://github.com/nextcloud/server/assets/1374172/42b54fd0-0a54-4361-819d-08bf7fd9a4f6

### Steps to reproduce

1. Create a folder
2. Click on its name with your mouse

### Expected behavior

Black border only shows if I navigate by keyboard

### Installation method

None

### Nextcloud Server version

28

### Operating system

None

### PHP engine version

None

### Web server

None

### Database engine version

None

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

_No response_

### List of activated Apps

_No response_

### Nextcloud Signing status

_No response_

### Nextcloud Logs

_No response_

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by ShGKme**
## Summary

* Fix #41490

There is an outline on `focus-visible` state, which appears also on click making a glitch effect.

At the same time, `focus-visible` state was not useful because the link was not focusable.

## Todo

- [x] Show outline only on real focus state
- [x] Make the link focusable by `tabindex` (links usually should not be focusable, but in this case it works close to the button)
- [x] Refactor `linkTo`: replace deprecated `is` attr on `<a>` with a dynamic component:
   - Special attribute `is` is deprecated and removed in Vue 3
   - It is confusing, that `<a>` element is rendered as `<span>` sometimes

Before | After
---|---
**Click** |
click-before | click-after
**Tab** |
tab-before | tab-after

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by jancborchardt — APPROVED** [review]
Looks good from the videos, thanks! :)

**Review by JuliaKirschenheuter — APPROVED** [review]

File: apps/files/src/components/FileEntry/FileEntryName.vue [hunk 1]
```diff
@@ -40,6 +40,7 @@
 	<a v-else
```
[the remark this answers is not in the document]
Comment (ShGKme) [reply apps/files/src/components/FileEntry/FileEntryName.vue:40]: It has `href` set by `v-bind="linkTo"`.

But after this comment I found that actually this `<a>` doesn't always render `<a>`

**Comment by ShGKme** [comment]
Updates:
1. Set `tabindex` in `linkTo` computed, not on `<a>` in template, because `<a>` is not guaranteed to be rendered as `<a>` here.
2. Refactor `linkTo`: replace deprecated `is` attr on `<a>` with a dynamic component:
   - Special attribute `is` is deprecated and removed in Vue 3
   - It is confusing, that `<a>` element is rendered as `<span>` sometimes

**Review by nimishavijay — APPROVED** [review]
Nice! Looks great design-wise!

**Review by marcoambrosini — APPROVED** [review]

**Comment by ShGKme** [comment]
Rebased onto master.

**Pull request merged**